hyperdome_server/web/share_mode.py

```diff
@@ -65,123 +66,114 @@ def __init__(self, common, web):
 
         @self.web.app.errorhandler(Exception)
         def unhandled_exception(e):
             e_str = ''.join(traceback.format_exception(type(e),
                                                        e,
                                                        e.__traceback__))
             print(e_str)
-            return e_str
+            return "Exception raised", 500
+
+        @self.web.app.route("/probe")
+        def probe():
+            self.info['online'] = str(len(self.counselors_available))
+            return json.dumps(self.info)
 
-        @self.web.app.route("/request_therapist", methods=['POST'])
-        def request_therapist():
+        @self.web.app.route("/request_counselor", methods=['POST'])
+        def request_counselor():
             guest_id = request.form['guest_id']
-            if self.therapists_available:
-                chosen_therapist = random.choice(self.therapists_available)
-                self.therapists_available.remove(chosen_therapist)
-                self.connected_therapist[guest_id] = chosen_therapist.username
-                self.connected_guest[chosen_therapist.username] = guest_id
-                return chosen_therapist.username
-            return ''
-
-        @login_required
-        @self.web.app.route("/therapy_complete", methods=['POST'])
-        def therapy_complete():
-            self.connected_therapist.pop(
-                self.connected_guest[current_user.username])
-            self.connected_guest.pop(current_user.username)
-            self.therapists_available.append(current_user)
-
-        @self.web.app.route("/therapist_signout", methods=["POST"])
-        @login_required
-        def therapist_signout():
-            user = load_user(request.form['username'])
-            logout_user(user)
-            self.therapists_available.remove(user)
-
-        @self.web.app.route("/therapist_signup", methods=["POST"])
-        def therapist_signup():
-            if request.form.get('masterkey', "") == "megumin":
-                if load_user(request.form['username']):
-                    return "Username already exists"
-                user = self.user_class(username=request.form['username'],
-                                       password=request.form['password'])
-                self.db.session.add(user)
-                self.db.session.commit()
-                return "Success"
-            return abort(401)
-
-        @self.web.app.route("/therapist_signin", methods=["POST"])
-        def therapist_signin():
-            # TODO authenticate
-            user = load_user(request.form['username'])
-            self.therapists_available.append(user)
+            counselors = [
+                counselor for counselor in self.counselors_available if self.counselors_available[counselor] > 0]
+            if not counselors:
+                return ''
+            chosen_counselor = random.choice(counselors)
+            self.counselors_available[chosen_counselor] -= 1
```

Comment:
thonk

Why not just use True and False? I'm not sure I follow

Reply:
My initial thought was that counselors might be able to take on multiple guests at a time. I'm not sure that I'm committed to that, and bool would be more obvious for the current system, but I'm going to stick with it for now.

Reply:
Ahh, I gotcha. Yeah, nothing wrong with keeping it as-is